### PR TITLE
fix(compression): Changes that follow core breaking changes to the confighttp API

### PR DESCRIPTION
**Pull request by rnishtala-sumo**
Changes that follow core breaking changes to the confighttp API

Link to tracking issue
Issue - https://github.com/open-telemetry/opentelemetry-collector/issues/10467

### CODE REVIEW

File: exporter/splunkhecexporter/go.mod [hunk 1]
```diff
@@ -135,3 +135,7 @@ replace github.com/open-telemetry/opentelemetry-collector-contrib/pkg/pdatautil
 replace github.com/open-telemetry/opentelemetry-collector-contrib/pkg/pdatatest => ../../pkg/pdatatest
 
 replace github.com/open-telemetry/opentelemetry-collector-contrib/pkg/golden => ../../pkg/golden
+
+replace go.opentelemetry.io/collector/config/configcompression => github.com/rnishtala-sumo/opentelemetry-collector/config/configcompression v0.0.0-20241118202008-518d136b719c
```
Comment (rnishtala-sumo) [inline exporter/splunkhecexporter/go.mod:139]: @dmitryax -  as requested, I am using core changes from my fork in contrib temporarily.

Comment (rnishtala-sumo) [reply exporter/splunkhecexporter/go.mod:139]: it isn't clear why I still get these errors, don't see them on my local.

`../../exporter/splunkhecexporter/client.go:660:54: undefined: configcompression.TypeWithLevel`

File: exporter/splunkhecexporter/go.mod [hunk 2]
```diff
@@ -137,3 +137,7 @@ replace github.com/open-telemetry/opentelemetry-collector-contrib/pkg/pdatautil
 replace github.com/open-telemetry/opentelemetry-collector-contrib/pkg/pdatatest => ../../pkg/pdatatest
 
 replace github.com/open-telemetry/opentelemetry-collector-contrib/pkg/golden => ../../pkg/golden
+
+replace go.opentelemetry.io/collector/config/configcompression => github.com/rnishtala-sumo/opentelemetry-collector/config/configcompression v0.0.0-20241204155200-78d83fd4ebca
```
Comment (atoulme) [inline exporter/splunkhecexporter/go.mod:141]: I would rather we merge the changes in core and use the latest main snapshot.

Comment (rnishtala-sumo) [reply exporter/splunkhecexporter/go.mod:141]: Closing this PR in favor of - https://github.com/open-telemetry/opentelemetry-collector/pull/11805

**Pull request closed without merging**